### PR TITLE
Add a warning for invalid string sequences

### LINKED ISSUE #688: Fix string escapes
See https://github.com/bazelbuild/starlark/issues/38 for the specification change.

### PR DESCRIPTION
**Pull request by vladmos**
Fixes #688.

### CODE REVIEW

File: warn/warn_operation.go [hunk 1]
```diff
@@ -108,3 +111,85 @@ func integerDivisionWarning(f *build.File) []*LinterFinding {
...
+			value = str.Token[3 : len(str.Token)-3]
+		} else {
+			quotes = str.Token[:1]
+			value = str.Token[1 : len(str.Token)-1]
+		}
+
+		var problems []int // positions of the problems
+
+		escaped := false
+		// This for-loop doesn't correclty check for a backlash at the end of the string literal, but
```
Comment (laurentlb) [inline warn/warn_operation.go]: correctly

File: warn/warn_operation.go [hunk 2]
```diff
@@ -108,3 +111,85 @@ func integerDivisionWarning(f *build.File) []*LinterFinding {
...
+		for i, ch := range value {
+			if !escaped {
+				if ch == '\\' {
+					escaped = true
+				}
+				continue
+			}
+
+			switch ch {
+			case '\n', '\\', 'n', 'r', 't', 'x', '\'', '"', '0', '1', '2', '3', '4', '5', '6', '7':
```
Comment (laurentlb) [inline warn/warn_operation.go]: Where does the list come from?
Can you include a link?

File: warn/warn_operation.go [hunk 3]
```diff
@@ -108,3 +111,85 @@ func integerDivisionWarning(f *build.File) []*LinterFinding {
...
+			escaped = false
+		}
+
+		if len(problems) == 0 {
+			return
+		}
+
+		var msg string
+		if len(problems) == 1 {
+			msg = fmt.Sprintf("Invalid quote sequences at position %d.", problems[0])
```
Comment (laurentlb) [inline warn/warn_operation.go]: Users might be confused by the message. Include at least the character that's incorrectly escaped.

Suggested:
`Invalid quote sequence '\w' at position 18.`

File: warn/warn_operation.go [hunk 4]
```diff
@@ -108,3 +111,85 @@ func integerDivisionWarning(f *build.File) []*LinterFinding {
...
+		} else {
+			msg = fmt.Sprintf(
+				"Invalid quote sequences at positions %s.",
+				strings.Trim(strings.Join(strings.Fields(fmt.Sprint(problems)), ", "), "[]"),
```
Comment (laurentlb) [inline warn/warn_operation.go]: Suggested:
```
Invalid quote sequences:
  '\w' at position 18
  '\z' at position 23
```
or:

`Invalid quote sequences '\w', '\z' at positions 18, 23`

File: warn/warn_operation.go (hunk 2 above)
Comment (vladmos) [reply warn/warn_operation.go]: Done.

File: warn/warn_operation.go (hunk 3 above)
Comment (vladmos) [reply warn/warn_operation.go]: Done.

File: warn/warn_operation.go (hunk 4 above)
Comment (vladmos) [reply warn/warn_operation.go]: Done (multiline warnings).

File: warn/warn_operation.go (hunk 1 above)
Comment (vladmos) [reply warn/warn_operation.go]: Fixed.

**Pull request merged**